src/main/kotlin/org/jitsi/rtp/rtcp/rtcpfb/payload_specific_fb/RtcpFbRembPacket.kt

```diff
@@ -0,0 +1,189 @@
+ * @author Boris Grozev
+ */
+class RtcpFbRembPacket(
+    buffer: ByteArray,
+    offset: Int,
+    length: Int
+) : PayloadSpecificRtcpFbPacket(buffer, offset, length) {
+
+    /**
+     * The exponential scaling of the mantissa for the maximum total media
+     * bit rate value, ignoring all packet overhead.
+     */
+    val exp: Int
+        get() = getBrExp(buffer, offset)
+
+    /**
+     * The mantissa of the maximum total media bit rate (ignoring all packet
+     * overhead) that the sender of the REMB estimates.  The BR is the estimate
+     * of the traveled path for the SSRCs reported in this message.
+     */
+    val mantissa: Int
+        get() = getBrMantissa(buffer, offset)
+
+    val bitrate: Long
+        get() = getBitrate(buffer, offset)
+
+    val numSsrc: Int
+        get() = getNumSsrc(buffer, offset)
+
+    /**
+     * one or more SSRC entries which this feedback message applies to.
+     */
+    val ssrcs: List<Long> by lazy {
+        (0 until numSsrc).map {
+            getSsrc(buffer, offset, it)
+        }.toList()
+    }
+
+    override fun clone(): RtcpFbRembPacket {
+        return RtcpFbRembPacket(buffer.cloneFromPool(), offset, length)
+    }
+
+    companion object {
+        const val FMT = 15
+
+        const val REMB_OFF = FCI_OFFSET
+        const val REMB_LEN = 4
+        const val NUM_SSRC_OFF = FCI_OFFSET + REMB_LEN
+        const val NUM_SSRC_LEN = 1
+        const val BR_OFF = NUM_SSRC_OFF + NUM_SSRC_LEN
+        const val BR_LEN = 4
+        const val SSRCS_OFF = NUM_SSRC_OFF + BR_LEN
+
+        fun getBrExp(buf: ByteArray, baseOffset: Int): Int =
+            buf.getBitsAsInt(baseOffset + BR_OFF, 0, 6)
+        fun getBrMantissa(buf: ByteArray, baseOffset: Int): Int =
+            (buf.getBitsAsInt(baseOffset + BR_OFF, 6, 2) shl 2) + buf.getShortAsInt(baseOffset + BR_OFF + 1)
+        fun getBitrate(buf: ByteArray, baseOffset: Int): Long =
+            (getBrMantissa(buf, baseOffset) * Math.pow(2.0, getBrExp(buf, baseOffset).toDouble())).toLong()
+        fun getNumSsrc(buf: ByteArray, baseOffset: Int): Int =
+            buf.getByteAsInt(baseOffset + NUM_SSRC_OFF)
+        fun getSsrc(buf: ByteArray, baseOffset: Int, ssrcIndex: Int) =
+            buf.getIntAsLong(baseOffset + SSRCS_OFF + ssrcIndex * 4)
+
+        fun getExpAndMantissa(brBps: Long): Pair<Int, Int> {
+            // 6 bit Exp
+            // 18 bit mantissa
+            var exp = 0
+            for (i in 0..63) {
+                if (brBps <= 0x3ffff shl i) {
+                    exp = i
+                    break
+                }
+            }
+
+            /* type of bitrate is an unsigned int (32 bits) */
+            val mantissa = brBps.toInt() shr exp
+            return Pair(exp, mantissa)
+        }
+    }
+}
+
+class RtcpFbRembPacketBuilder(
+    val rtcpHeader: RtcpHeaderBuilder = RtcpHeaderBuilder(),
+    val ssrcs: List<Long> = emptyList(),
+    val exp: Int,
+    val mantissa: Int
+) {
+
+    constructor(
```

Comment:
do we need all 3 constructors for different use cases, or is it just used to parse the fields?

Reply:
might be easier to just make the values `vars` with some default and then set them, since this is a builder.